### PR TITLE
Add security to atop connector

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -112,6 +112,7 @@
         <module>presto-docs</module>
         <module>presto-verifier</module>
         <module>presto-testing-server-launcher</module>
+        <module>presto-base-plugin</module>
```
Comment (dain) [inline pom.xml]: @martint and @electrum should review this name

File: presto-base-plugin/src/main/java/com/facebook/presto/plugin/base/security/FileBasedAccessControl.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+    @Inject
+    public FileBasedAccessControl(FileBasedAccessControlConfig config, JsonCodec<AccessControlRules> codec)
+    {
+        AccessControlRules rules;
+        try {
+            rules = codec.fromJson(Files.readAllBytes(Paths.get(config.getConfigFile())));
+        }
+        catch (IOException e) {
+            throw Throwables.propagate(e);
```
Comment (dain) [inline presto-base-plugin/src/main/java/com/facebook/presto/plugin/base/security/FileBasedAccessControl.java]: why not just throw `IOException` from this method?

**Comment by dain** [comment]
I'd move the `ReadOnlyAccessControl` here also

File: presto-atop/src/main/java/com/facebook/presto/atop/ConditionalModule.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+package com.facebook.presto.atop;
+
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.airlift.configuration.ConfigurationAwareModule;
+
+import java.util.function.Predicate;
+
+public class ConditionalModule<T>
```
Comment (dain) [inline presto-atop/src/main/java/com/facebook/presto/atop/ConditionalModule.java]: We should move this to airlift

**Comment by dain** [comment]
Other than the module name discussion, looks good

File: pom.xml (hunk 1 above)
Comment (martint) [inline pom.xml]: How about presto-plugin-toolkit?

File: presto-atop/src/main/java/com/facebook/presto/atop/ConditionalModule.java (hunk 3 above)
Comment (cberner) [inline presto-atop/src/main/java/com/facebook/presto/atop/ConditionalModule.java]: Good idea. I'll do that in a separate commit

File: presto-base-plugin/src/main/java/com/facebook/presto/plugin/base/security/FileBasedAccessControl.java (hunk 2 above)
Comment (cberner) [inline presto-base-plugin/src/main/java/com/facebook/presto/plugin/base/security/FileBasedAccessControl.java]: Oh ya, good idea

File: pom.xml (hunk 1 above)
Comment (cberner) [inline pom.xml]: Works for me

**Comment by cberner** [comment]
"I'd move the ReadOnlyAccessControl here also". Will do that in a separate commit also, since I'm going to add file security to the Hive connector so will do it at the same time

**Pull request merged**